libpysal/graph/base.py

```diff
@@ -1419,6 +1421,39 @@ def eliminate_zeros(self):
         zeros = (self._adjacency == 0) != isolates
         return Graph(self._adjacency[~zeros], is_sorted=True)
 
+    def assign_self_weight(self, weight=1):
+        """Assign values to edges representing self-weight.
+
+        The value for each ``focal == neighbor`` location in the graph is set to ``weight``.
```

Comment:
Looks like `pre-commit` is complaining that this line is too long.

```suggestion
        The value for each ``focal == neighbor`` location in
        the graph is set to ``weight``.
```